awscli/customizations/cloudformation/deploy.py

```diff
@@ -62,6 +65,40 @@ class DeployCommand(BasicCommand):
             )
         },
         {
+            'name': 's3-bucket',
+            'required': False,
+            'help_text': (
+                'The name of the S3 bucket where this command uploads'
```

Comment:
I would expand help string to precisely explain when you upload to S3, and how it is optional.